var/spack/repos/builtin/packages/hdf/package.py

```diff
@@ -195,14 +198,13 @@ def _test_list(self):
         """This test compares low-level HDF file information to expected."""
         storm_fn = os.path.join(self.install_test_root,
                                 self.extra_install_tests, 'storm110.hdf')
+        test_data_dir = self.test_suite.current_test_data_dir
         work_dir = '.'
 
-        reason = 'test: ensure hdfls produces expected output'
-        details_file = os.path.join(self.test_dir.data.hdf, 'storm110.out')
-        output = ''
-        with open(details_file, 'r') as fd:
-            output == fd.read()
-        self.run_test('hdfls', [storm_fn], output, installed=True,
+        reason = 'test: ensuring hdfls produces expected output'
+        details_file = os.path.join(test_data_dir, 'storm110.out')
+        expected = sit.get_expected_output(details_file)
```

Comment:
I think we should push this into module scope for the package as part of `setup_package`. Do you want to do this as part of this PR or we can schedule it for a separate one.

Reply:
As discussed on spack slack, this will be a separate PR

Reply:
Done in #18339